### PR TITLE
FIX Remove legacy sprintf style translations in favour of placeholders

### PR DESCRIPTION
**Pull request by robbieaverill**
Issue: https://github.com/silverstripe/silverstripe-framework/issues/6582

Note: CMSBatchAction_* classes have still got `%d` placeholders in them, because they get passed around a bit before being replaced after the actual translation. I figured this isn't appropriate to replace at this point as it's logic change rather than sprintf -> named placeholder swapping.

### CODE REVIEW

**Comment by dhensby** [comment]
big conflicts

**Comment by tractorcow** [comment]
@robbieaverill can we rebase this and I'll merge it in.

**Comment by robbieaverill** [comment]
@tractorcow rebased, thanks =)

File: code/Controllers/CMSMain.php [hunk 1]
```diff
@@ -1742,7 +1742,7 @@ public function delete($data, $form)
 
         $this->getResponse()->addHeader(
             'X-Status',
-            rawurlencode(sprintf(_t('SilverStripe\\CMS\\Controllers\\CMSMain.REMOVEDPAGEFROMDRAFT', "Removed '%s' from the draft site"), $record->Title))
+            rawurlencode(sprintf(_t(__CLASS__ . '.REMOVEDPAGEFROMDRAFT', "Removed '{title}' from the draft site"), $record->Title))
```
Comment (tractorcow) [inline code/Controllers/CMSMain.php]: Darn, I just realised we need to remove these sprintf calls.

Comment (robbieaverill) [inline code/Controllers/CMSMain.php]: Fixed in #1915

**Comment by tractorcow** [comment]
I've manually cherry picked this to 4 branch.

**Pull request merged**